python/ray/dataframe/dataframe.py

```diff
@@ -713,9 +724,25 @@ def dropna(self, axis, how, thresh=None, subset=[], inplace=False):
         raise NotImplementedError("Not yet")
 
     def add(self, other, axis='columns', level=None, fill_value=None):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Add this DataFrame to another or a scalar/list.
+
+        Args:
+            other: What to add this this DataFrame.
+            axis: The axis to apply addition over. Only applicaable to Series or list 'other'.
+            level: A level in the multilevel axis to add over.
+            fill_value: The value to fill NaN.
+
+        Returns:
+            A new DataFrame with the applied addition.
+        """
+        if isinstance(other, DataFrame):
+            return self._inter_df_op_helper(
+                lambda x, y: x.add(y, axis, level, fill_value),
+                other, axis, level)
+        else:
+            return self._single_df_op_helper(
+                lambda df: df.add(other, axis, level, fill_value),
```

Comment:
Please note here (and everywhere else) that this can cause implicit serialization issues if `other` is a series, for the eventual refactor